src/operators/resample.ml

```diff
@@ -30,6 +30,8 @@ object (self)
   (* Hide our child: we'll treat it specially. *)
   inherit source ~name:"resample" kind as super
 
+  method self_sync = source#self_sync
```

Comment:
Ouch this one is really tricky: we are changing the speed of the source so that if it is self-syncing then we will have slower time as output. But I guess this actually does not cause any problem? Slower time is a valid notion of time?

Reply:
Good remark, thanks. The code below:
```ocaml
  method private set_clock =
    let c = Clock.create_known (new Clock.clock self#id) in
      Clock.unify
        self#clock (Clock.create_unknown ~sources:[] ~sub_clocks:[c]) ;
      Clock.unify source#clock c ;
      (* Make sure the slave clock can be garbage collected, cf. cue_cut(). *)
      Gc.finalise (fun self -> Clock.forget self#clock c) self
```
Assures that the clock is the same as `source#clock`. Thence, it makes sense to also have the same `self_sync`.

If `source` has its own definition of time, the operator will still work but not as expected. It would be nice to catch this statically and fail if the underlying source cannot be accelerated or slowed down but I don't think that this is possible at the moment (in particular because this is a dynamic feature). The same limitation applies to sources used in crossfade transitions.